nixos/modules/services/web-apps/lasuite-docs.nix

```diff
@@ -0,0 +1,473 @@
+          Configure local Redis cache server for docs.
+        '';
+      };
+    };
+
+    collaborationServer = {
+      package = mkPackageOption pkgs "lasuite-docs-collaboration-server" { };
+
+      port = mkOption {
+        type = types.port;
+        default = 4444;
+        description = ''
+          Port used by the collaboration server to listen.
+        '';
+      };
+
+      serverSecretPath = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = ''
+          Path to the server secret of the collaboration server.
+        '';
+      };
+
+      yproviderApiKeyPath = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = ''
+          Path to the y-provider API key.
+        '';
+      };
+
+      config = mkOption {
+        type = types.attrsOf (
+          types.oneOf [
+            types.str
+            types.bool
+          ]
+        );
+        default = { };
+        example = ''
+          {
+            COLLABORATION_LOGGING = true;
+          }
+        '';
+        description = ''
+          Configuration options of collaboration server.
+
+          See https://github.com/suitenumerique/docs/blob/v3.1.0/docs/env.md
+        '';
+      };
+    };
+
+    domain = mkOption {
+      type = types.str;
+      description = ''
+        Domain name of the docs instance.
+      '';
+    };
+
+    config = mkOption {
+      type = types.attrsOf (
+        types.oneOf [
+          types.str
+          types.bool
+        ]
+      );
+      default = { };
+      example = ''
+        {
+          DJANGO_ALLOWED_HOSTS = "*";
+        }
+      '';
+      description = ''
+        Configuration options of docs.
+
+        See https://github.com/suitenumerique/docs/blob/v3.1.0/docs/env.md
+      '';
+    };
+
+    environmentFile = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = ''
+        Path to environment file.
+
+        This can be useful to pass secrets to docs via tools like `agenix` or `sops`.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    services.lasuite-docs.config =
+      {
+        DJANGO_CONFIGURATION = mkDefault "Production";
+        DJANGO_SETTINGS_MODULE = mkDefault "impress.settings";
+        DATA_DIR = mkDefault "/var/lib/lasuite-docs";
+      }
+      // (optionalAttrs cfg.enableNginx {
+        DJANGO_ALLOWED_HOSTS = mkDefault "localhost,127.0.0.1,${cfg.domain}";
+      })
+      // (optionalAttrs cfg.database.createLocally {
+        DB_NAME = mkDefault "lasuite-docs";
+        DB_USER = mkDefault "lasuite-docs";
+        DB_HOST = mkDefault "/run/postgresql";
+      })
+      // (optionalAttrs cfg.redis.createLocally {
+        REDIS_URL = mkDefault "unix://${config.services.redis.servers.lasuite-docs.unixSocket}?db=1";
+        CELERY_BROKER_URL = mkDefault "redis+socket://${config.services.redis.servers.lasuite-docs.unixSocket}?db=2";
+      });
```

Comment:
These should become options on `services.lasuite-docs.settings` instead, so they're discoverable. For the ones that can't be changed they can be made read-only or even internal.